tajo-plan/src/main/java/org/apache/tajo/plan/joinorder/GreedyHeuristicJoinOrderAlgorithm.java

```diff
@@ -484,7 +487,7 @@ public static double getCost(LogicalNode node) {
     default:
       // all binary operators (join, union, except, and intersect) are handled in the above cases.
       // So, we need to handle only unary nodes in default.
-      cost = getCost(((UnaryNode) node).getChild());
+      cost = getCost((LogicalNode)((UnaryNode) node).getChild());
```

Comment:
Basically, It cannot be avoidable. The problem is that both getChild() and getCost() require type inference.